packages/core/src/server/service.ts

```diff
@@ -160,19 +167,46 @@ export class ServerService {
           data: undefined,
           errors,
         };
-        return res.status(503).json(result);
+        return response.status(503).json(result);
       }
 
-      return graphqlMiddleware(req, res, next);
+      switch (request.method) {
+        case "POST":
+          return graphqlMiddleware(request, response, next);
+        case "GET": {
+          const SERVER_URL = this.#getOwnUrlFromRequest(request);
+          return response
+            .status(200)
+            .setHeader("Content-Type", "text/html")
+            .send(graphiQLHtml({ endpoint: `${SERVER_URL}/graphql` }));
+        }
+        case "HEAD":
+          return response.status(200).send();
```

Comment:
Any reason to specify a behavior for `"HEAD"` here rather than just falling back to the default behavior offered by express (via `next()`) ?

Reply:
`HEAD` request errors out when it's left out in both `/` and `/graphql` but not in the others (`/metrics`, etc.). I don't understand why but that's why I'm explicitly adding it here. My guess is it has something to do with `.use(...)` vs `.get`/`.<method>`? Not sure